addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/WebCacheDeceptionScanRule.java

```diff
@@ -87,12 +85,15 @@ public void scan() {
                     uri.setPath(newPath);
                     authorisedMessage.getRequestHeader().setURI(uri);
                     sendAndReceive(authorisedMessage);
-                    String cachedResponse = authorisedMessage.getResponseBody().toString();
-                    String unauthorisedCachedResponse =
+                    ComparableResponse authCompResp =
+                            new ComparableResponse(authorisedMessage, null);
+                    HttpMessage unauthorisedCachedResponse =
                             makeUnauthorisedRequest(
                                     uri, authorisedMessage.getRequestHeader().getMethod());
-
-                    if (checkSimilarity(unauthorisedCachedResponse, cachedResponse)) {
+                    ComparableResponse unAuthCompResp =
+                            new ComparableResponse(unauthorisedCachedResponse, null);
+                    float similarity = authCompResp.compareWith(unAuthCompResp);
+                    if (similarity >= 0.75) {
```

Comment:
This is a made up threshold. All of the tests seem to pan out to 1.0 or 0.0, and I didn't see an easy way to appropriately make partial matches. (I probably just didn't look hard enough :cry:)

Reply:
@bettercalln1ck any thoughts on either what a good match threshold would be (0 being totally different and 1.0 being very similar)? Or how to appropriately cause partial matches in the UnitTests?

Reply:
I've tweaked this to 80 after looking at some other things over the weekend.